dex/farm/src/claim_progress.rs

```diff
@@ -0,0 +1,114 @@
+elrond_wasm::imports!();
+
+use common_errors::{ERROR_DIFFERENT_ATTRIBUTES_FOR_MERGE, ERROR_EMPTY_PAYMENTS};
+use common_structs::{Nonce, PaymentsVec};
+use weekly_rewards_splitting::ClaimProgress;
+
+#[elrond_wasm::module]
+pub trait ClaimProgressModule:
+    farm_boosted_yields::FarmBoostedYieldsModule
+    + permissions_module::PermissionsModule
+    + energy_query::EnergyQueryModule
+    + week_timekeeping::WeekTimekeepingModule
+    + weekly_rewards_splitting::events::WeeklyRewardsSplittingEventsModule
+    + weekly_rewards_splitting::WeeklyRewardsSplittingModule
+{
+    // Clears the claim progress for all the additional payment tokens
+    // in order to be able to merge the users claim progress under a single token nonce
+    fn clear_payments_claim_progress(
+        &self,
+        user: &ManagedAddress,
+        all_payments: &PaymentsVec<Self::Api>,
+    ) -> Nonce {
+        require!(!all_payments.is_empty(), ERROR_EMPTY_PAYMENTS);
+        if all_payments.len() > 1 {
+            let mut additional_payments = all_payments.clone();
+            let first_payment = additional_payments.get(0);
+            additional_payments.remove(0);
+            let first_claim_progress =
+                self.get_claim_progress_or_default(user, first_payment.token_nonce);
+            let first_payment_claim_week = first_claim_progress.week;
+            let first_payment_energy = first_claim_progress.energy;
+            for payment in additional_payments.iter() {
+                let payment_claim_progress =
+                    self.get_claim_progress_and_clear(user, payment.token_nonce);
+                require!(
+                    first_payment_claim_week == payment_claim_progress.week
+                        && first_payment_energy == payment_claim_progress.energy,
+                    ERROR_DIFFERENT_ATTRIBUTES_FOR_MERGE
+                );
```

Comment:
It's probably better to derive `PartialEq` for `ClaimProgress` struct than manually compare here.

Reply:
Done.